File: dinner-ideas/src/models/DinnerItem.ts
import { BaseItem } from "./BaseItem";
import { FoodTag } from "./FoodTag";
import { Ingredient } from "./Ingredient";

export interface DinnerItem extends BaseItem {
    name: string;
    description: string;
    prepTime: number;
    cookTime: number;
    steps: DinnerItemStep[];
    tags: FoodTag[];
    ingredients: Ingredient[];
    imageKey?: string;
}

export interface DinnerItemStep {
    stepTitle: string;
    stepDescription: string;
    id: string;
    /** Denormalised list of ingredient IDs used in this step. */
    ingredientIds?: string[];
}

File: dinner-ideas-api/dinner-ideas-lambda/models/DinnerItem.cs
namespace dinner_ideas_lambda.models;

public class DinnerItem : BaseItem
{
    public override string TypeAndId { get => $"{nameof(DinnerItem)}|{Id}"; }
    public required string Name { get; set; }
    public required string Description { get; set; }
    public required int PrepTime { get; set; }
    public required int CookTime { get; set; }
    public DinnerItemStep[] Steps { get; set; } = [];
    public FoodTag[] Tags { get; set; } = [];
    public int TotalTime => PrepTime + CookTime;

}